### ISSUE TITLE
Phone can't record voicemails

### ISSUE THREAD

**Issue by purpleidea** [stefan-langenmaier/ring-overlay#63]
So I'm not sure if this is in the right place... If there's a better upstream for sflphone, lmk. Otherwise here maybe?

I used one of the core sflphone packages in Fedora 30. I installed sflphone and got it registered with a FreePBX/asterisk server. Current and up-to-date as of today.

Long story short, whenever I would use it to leave a voicemail on someone's extension, it wouldn't work. Some logs:

```
The sflphone sucks.

[2019-11-27 03:08:03] VERBOSE[4527][C-00000003] pbx_builtins.c: Goto (macro-vm,s-NOANSWER,1)
[2019-11-27 03:08:03] VERBOSE[4527][C-00000003] pbx.c: Executing [s-NOANSWER@macro-vm:1] Macro("PJSIP/1002-00000004", "get-vmcontext,100
1") in new stack
[2019-11-27 03:08:03] VERBOSE[4527][C-00000003] pbx.c: Executing [s@macro-get-vmcontext:1] Set("PJSIP/1002-00000004", "VMCONTEXT=default
") in new stack
[2019-11-27 03:08:03] VERBOSE[4527][C-00000003] pbx.c: Executing [s@macro-get-vmcontext:2] GotoIf("PJSIP/1002-00000004", "0?200:300") in
 new stack
[2019-11-27 03:08:03] VERBOSE[4527][C-00000003] pbx_builtins.c: Goto (macro-get-vmcontext,s,300)
[2019-11-27 03:08:03] VERBOSE[4527][C-00000003] pbx.c: Executing [s@macro-get-vmcontext:300] NoOp("PJSIP/1002-00000004", "") in new stac
k
[2019-11-27 03:08:03] VERBOSE[4527][C-00000003] pbx.c: Executing [s-NOANSWER@macro-vm:2] VoiceMail("PJSIP/1002-00000004", "1001@default,
u") in new stack
[2019-11-27 03:08:04] VERBOSE[4527][C-00000003] file.c: <PJSIP/1002-00000004> Playing '/var/spool/asterisk/voicemail/default/1001/unavai
l.slin' (language 'en')  

[2019-11-27 03:08:07] VERBOSE[4527][C-00000003] file.c: <PJSIP/1002-00000004> Playing 'vm-intro.ulaw' (language 'en')

[2019-11-27 03:08:13] VERBOSE[4527][C-00000003] file.c: <PJSIP/1002-00000004> Playing 'beep.ulaw' (language 'en')
[2019-11-27 03:08:13] VERBOSE[4527][C-00000003] app_voicemail.c: Recording the message
[2019-11-27 03:08:13] VERBOSE[4527][C-00000003] app.c: x=0, open writing:  /var/spool/asterisk/voicemail/default/1001/tmp/zAcM3R format:
 wav, 0x7fa0c0014f40

[2019-11-27 03:08:22] VERBOSE[4527][C-00000003] app.c: User ended message by pressing #
[2019-11-27 03:08:22] VERBOSE[4527][C-00000003] file.c: <PJSIP/1002-00000004> Playing 'auth-thankyou.ulaw' (language 'en')
[2019-11-27 03:08:23] VERBOSE[4527][C-00000003] app_voicemail.c: Recording was 0 seconds long but needs to be at least 1 - abandoning
[2019-11-27 03:08:23] VERBOSE[4527][C-00000003] pbx.c: Executing [s-NOANSWER@macro-vm:3] Goto("PJSIP/1002-00000004", "exit-FAILED,1") in
 new stack
[2019-11-27 03:08:23] VERBOSE[4527][C-00000003] pbx_builtins.c: Goto (macro-vm,exit-FAILED,1)
[2019-11-27 03:08:23] VERBOSE[4527][C-00000003] pbx.c: Executing [exit-FAILED@macro-vm:1] Playback("PJSIP/1002-00000004", "im-sorry&an-e
rror-has-occurred") in new stack
[2019-11-27 03:08:23] VERBOSE[4527][C-00000003] file.c: <PJSIP/1002-00000004> Playing 'im-sorry.ulaw' (language 'en')
[2019-11-27 03:08:24] VERBOSE[4527][C-00000003] file.c: <PJSIP/1002-00000004> Playing 'an-error-has-occurred.ulaw' (language 'en')

```

Long story short, there's bugs. This `rfc2833compensate=yes` might be related, I didn't test.

See:

 https://community.freepbx.org/t/sip-no-audio-on-voicemail-from-inbound-calls-phones-work-fine-calles-transferred-to-voicemail-work-too/24179/3

and 

https://community.spiceworks.com/topic/2166075-freepbx-voicemail-issue-using-sip-uri-issue

Not sure how to fix this, but I figured I'd start by reporting it.

HTH!

**Comment by stefan-langenmaier** [comment]
Hello @purpleidea ,

first of all sorry that it took me so long to reply.
Well, this repo is  only for packaging Ring/Jami the successor of sflphone for Gentoo.
I know that it's still possible to configure sip accounts in Jami but unfortunately I have no expertise in this area.
I think the best starting point to get into contact with upstream is: https://jami.net/contribute/

Best regards,
Stefan
